### PR TITLE
Fetch selected NFT PFP from Moralis, without the overlay

### PR DESCRIPTION
**Pull request by udhriaznov**
## Background

We decided to try fetching the NFT metadata from the Moralis API instead of parsing it with our custom code in the metadata service, so that we could just leave the custom code for the overlay on the NFT

## Related Story / Issue

https://linear.app/unstoppable-domains/issue/ECUX-186/fetch-selected-nft-pfp-from-moralis-without-the-overlay

## Changes

- GET `/metadata/:domainOrToken`
  - Added Moralis SDK & started receiving NFT metadata from Moralis API
  - Improved the URL parsing for metadata image
  - Added `withOverlay` parameter to send image without the overlay (also, this endpoint works much more stable now)
  - Removed the unused code for NFT fetching that we used before the usage of Moralis API 
- GET `/image/:domainOrToken`
- GET `/image-src/:domainOrToken`
  - Made them work similar to GET `/metadata/:domainOrToken` endpoint so that now they return not only the default generated domain picture, but also the custom one, which was selected by user.

### CODE REVIEW

File: src/controllers/MetaDataController.ts [hunk 1]
```diff
@@ -116,37 +165,68 @@ export class MetaDataController {
...
+      fetchedMetadata = await response.json();
+      image = fetchedMetadata?.image;
+    }
     let socialPicture = '';
-    if (pictureOrUrl) {
+
+    if (!!image && withOverlay) {
       let data = '',
         mimeType = null;
       if (nftStandard === 'cryptopunks') {
```
Comment (sudoryan) [inline src/controllers/MetaDataController.ts:205]: no longer need to do custom resolving of cryptopunks thanks to API.
We can remove this whole (nftStandard === 'cryptopunks') {} statement

**Review by sudoryan — APPROVED** [review]
Great improvement! Much faster response times. Remove custom cryptopunk handling and we're good to go

**Review by iurevych — APPROVED** [review]
LGTM

File: src/utils/socialPicture/index.ts [hunk 2]
```diff
@@ -38,211 +16,27 @@ const parsePictureRecord = (avatarRecord: string) => {
...
-      );
-      break;
-
-    case Network.Binance:
-      provider = new ethers.providers.JsonRpcProvider(
-        env.APPLICATION.BINANCE.JSON_RPC_API_URL,
-      );
-      break;
+const makeImageLink = (imageUrl: string) => {
+  const PENATA_URL = 'https://gateway.pinata.cloud/ipfs/';
```
[the remark this answers is not in the document]
Comment (udhriaznov) [reply src/utils/socialPicture/index.ts:20]: oops, thanks

**Pull request merged**